src/components/Button/Button.jsx

```diff
@@ -41,8 +43,10 @@ Button.propTypes = {
   type: propTypes.oneOf(['button', 'submit', 'reset']),
   /** Any additional classes to be applied to the button */
   className: propTypes.string,
+  /** The icon shows on button */
+  icon: propTypes.oneOf(Object.keys(iconNames)),
   /** Children elements. */
-  children: propTypes.string,
+  children: propTypes.node,
```

Comment:
should we restrict this to only `string` types since we don't want people throwing markup inside buttons?